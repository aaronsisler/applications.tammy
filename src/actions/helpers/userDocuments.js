export const setUserDocuments = (userDocuments) => ({
    type: 'SET_USER_DOCUMENTS',
    userDocuments,
});

// export const addUserDocument = (userDocument) => ({
//     type: 'ADD_USER_DOCUMENT',
//     userDocument,
// });
