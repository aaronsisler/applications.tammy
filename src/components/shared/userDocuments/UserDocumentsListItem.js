import React from 'react';
import PropTypes from 'prop-types';

export default class UserDocumentsListItem extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <a className="document_list_item" href={`${this.props.downloadURL}`}>
                <div className="document_list_item__title">
                    {this.props.documentName}
                </div>
                <div className="document_list_item__date_uploaded">
                    {this.props.dateUploaded}
                </div>
            </a>
        );
    }
}

UserDocumentsListItem.propTypes = {
    downloadURL: PropTypes.string.isRequired,
    documentName: PropTypes.string.isRequired,
    dateUploaded: PropTypes.string.isRequired,
};
