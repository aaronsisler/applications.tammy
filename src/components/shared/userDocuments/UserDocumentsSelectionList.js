import React from 'react';
import PropTypes from 'prop-types';
import UserDocumentsSelectionListItem from './UserDocumentsSelectionListItem';

export default class UserDocumentsSelectionList extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        if (this.props.userDocuments.length === 0) {
            return (
                <div className="user_documents_selection_list empty">
                    No available items
                </div>
            );
        }

        const applicationUserDocumentIds = (
            this.props.applicationUserDocuments.map((applicationUserDocument) => applicationUserDocument.userDocumentId)
        );

        return (
            <div className="user_documents_selection_list">
                {
                    this.props.userDocuments.map((userDocument, index) => {
                        const isDocumentAdded = applicationUserDocumentIds
                            .indexOf(userDocument.userDocumentId) != -1;

                        return (
                            <UserDocumentsSelectionListItem
                                key={index}
                                isDocumentAdded={isDocumentAdded}
                                {...userDocument}
                            />
                        );
                    })
                }
            </div>
        );
    }
}

UserDocumentsSelectionList.propTypes = {
    applicationUserDocuments: PropTypes.array.isRequired,
    userDocuments: PropTypes.array.isRequired,
};
