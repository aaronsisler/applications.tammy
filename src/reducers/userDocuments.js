const userDocumentsReducerDefaultState = [];

const userDocumentsReducer = (state = userDocumentsReducerDefaultState, action) => {
    switch (action.type) {
        case 'SET_USER_DOCUMENTS':
            return action.userDocuments;
        default:
            return state;
    }
};

export default userDocumentsReducer;
