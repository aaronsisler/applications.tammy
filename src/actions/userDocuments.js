import database from '../firebase/firebase';
import { setUserDocuments } from 'Actions/helpers/userDocuments';

export const startSetUserDocuments = (userId) => (dispatch) =>
    database.ref(`user_documents/${userId}`).once('value').then((snapshot) => {
        const userDocuments = [];
        snapshot.forEach((childSnapshot) => {
            userDocuments.push({
                id: childSnapshot.key,
                ...childSnapshot.val(),
            });
        });

        dispatch(setUserDocuments(userDocuments));
    })
