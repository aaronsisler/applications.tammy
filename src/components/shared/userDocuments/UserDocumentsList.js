import React from 'react';
import PropTypes from 'prop-types';
import UserDocumentsListItem from './UserDocumentsListItem';

export default class UserDocumentsList extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        if (this.props.userDocuments.length === 0) {
            return (
                <div className="user_documents_list empty">
                    No available items
                </div>
            );
        }

        return (
            <div className="user_documents_list">
                {
                    this.props.userDocuments.map((userDocument, index) =>
                        <UserDocumentsListItem key={index} {...userDocument} />)
                }
            </div>
        );
    }
}

UserDocumentsList.propTypes = {
    userDocuments: PropTypes.array,
};
