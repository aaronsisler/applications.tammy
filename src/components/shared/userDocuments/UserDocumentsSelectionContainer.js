import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import UserDocumentsSelectionList from './UserDocumentsSelectionList';

export class UserDocumentsSelectionContainer extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <div className="user_documents_selection_container">
                <UserDocumentsSelectionList
                    applicationUserDocuments={this.props.applicationUserDocuments}
                    userDocuments={this.props.userDocuments}
                />
            </div>
        );
    }
}

/* istanbul ignore next */
const mapStateToProps = (state) => ({
    applicationUserDocuments: state.application.userDocuments,
    userDocuments: state.userDocuments,
});

export default connect(mapStateToProps)(UserDocumentsSelectionContainer);

UserDocumentsSelectionContainer.propTypes = {
    applicationUserDocuments: PropTypes.array.isRequired,
    userDocuments: PropTypes.array.isRequired,
};
