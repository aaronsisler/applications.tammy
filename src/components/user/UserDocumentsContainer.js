import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
// import UserDocumentsUploadWidget from '../shared/userDocuments/UserDocumentsUploadWidget';
import UserDocumentsList from '../shared/userDocuments/UserDocumentsList';

export class UserDocumentsContainer extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <div>
                {this.props.userDocuments &&
                    <div id="user_documents_container">
                        <UserDocumentsList
                            userDocuments={this.props.userDocuments}
                        />
                    </div>
                }
            </div>
        );
    }
}

const mapStateToProps = (state) => ({
    userDocuments: state.userDocuments,
})

export default connect(mapStateToProps)(UserDocumentsContainer);

UserDocumentsContainer.propTypes = {
    userDocuments: PropTypes.array,
};
