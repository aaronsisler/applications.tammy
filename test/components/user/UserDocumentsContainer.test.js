import React from 'react';
import { shallow } from 'enzyme';
import { UserDocumentsContainer } from 'User/UserDocumentsContainer';
import userDocuments from '../../fixtures/userDocuments';

describe('UserDocumentsContainer', () => {
    let wrapper;

    const buildWrapper = (userDocumentsInput = []) => {
        wrapper = shallow(
            <UserDocumentsContainer
                userDocuments={userDocumentsInput}
            />
        );
    };

    it('should render UserDocumentsContainer correctly when user documents are available', () => {
        buildWrapper(userDocuments);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render UserDocumentsContainer correctly when user documents are NOT available', () => {
        buildWrapper();

        expect(wrapper).toMatchSnapshot();
    });
});
