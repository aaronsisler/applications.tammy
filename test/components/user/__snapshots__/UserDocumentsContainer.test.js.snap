// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDocumentsContainer should render UserDocumentsContainer correctly when user documents are NOT available 1`] = `
<div>
  <div
    id="user_documents_container"
  >
    <DocumentList
      userDocuments={Array []}
    />
  </div>
</div>
`;

exports[`UserDocumentsContainer should render UserDocumentsContainer correctly when user documents are available 1`] = `
<div>
  <div
    id="user_documents_container"
  >
    <DocumentList
      userDocuments={
        Array [
          Object {
            "dateUploaded": "2018/01/01 01:23:45A",
            "documentName": "Document Name 1",
            "downloadURL": "downloadURL1",
          },
          Object {
            "dateUploaded": "2018/02/02 01:23:45A",
            "documentName": "Document Name 2",
            "downloadURL": "downloadURL2",
          },
        ]
      }
    />
  </div>
</div>
`;
